### PR TITLE
Feature/san 5536

### PR DESCRIPTION
**Pull request by GarageInc**
Done:
NEED TO MERGE ONLY AFTER FIXING San-5587 on backend (merged on stage)
* nightmode (selects from san-ui)
image
* icons for assets in form
image
* sharing of signals (if public signal)
image
image
image
image
image

### CODE REVIEW

File: src/ducks/Signals/signalFormManager/aboutForm/AboutForm.js [hunk 1]
```diff
@@ -38,74 +47,76 @@ const AboutForm = ({ triggerMeta, onSubmit, onBack }) => {
...
-      )}
+            <div className={styles.controls}>
+              <Button
+                type='button'
+                variant={'flat'}
+                accent='grey'
+                border
+                onClick={() => {
+                  onBack(trigger)
+                }}
```
Comment (haritonasty) [inline src/ducks/Signals/signalFormManager/aboutForm/AboutForm.js]: remove unnecessary brackets around `onBack(trigger)`

File: src/ducks/Signals/signalFormManager/aboutForm/AboutForm.js [hunk 2]
```diff
@@ -38,74 +47,76 @@ const AboutForm = ({ triggerMeta, onSubmit, onBack }) => {
...
+              >
+                Back
+              </Button>
+              <Button
+                type='submit'
+                disabled={!isValid || isSubmitting}
+                isActive={isValid && !isSubmitting}
+                variant={'fill'}
```
Comment (haritonasty) [inline src/ducks/Signals/signalFormManager/aboutForm/AboutForm.js]: we also can add `isLoading` prop if `isSubmitting = true`, maybe

File: src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormMetricValues.js [hunk 3]
```diff
@@ -41,140 +41,129 @@ export const TriggerFormMetricValues = ({
...
-          />
-          {isPriceMetric && isSingleTarget && (
-            <LastPriceComponent lastPrice={lastPrice} />
-          )}
-        </div>
-      )}
+    blocks &&
+    blocks.length > 0 && (
+      <div className={cx(styles.row, isTimeWindow ? styles.rowTimeWindow : '')}>
+        {type && blocks.includes('absoluteBorderRight') && (
```
Comment (haritonasty) [inline src/ducks/Signals/signalFormManager/signalCrudForm/formParts/TriggerFormMetricValues.js]: `isTimeWindow ? styles.rowTimeWindow : ' ' ` 
-> 
`isTimeWindow && styles.rowTimeWindow`

File: src/ducks/Signals/signalFormManager/signalMaster/SignalMaster.js [hunk 4]
```diff
@@ -1,183 +1,161 @@
...
-        trigger: {
-          ...newProps.trigger.trigger
-        },
-        step: newProps.step
-      })
+    case TRIGGER_STEPS.CONFIRM: {
+      if (isUpdate) {
+        return formData.isPublic
+          ? 'Update public signal'
+          : 'Update private signal'
```
Comment (haritonasty) [inline src/ducks/Signals/signalFormManager/signalMaster/SignalMaster.js]: to reduce repetitions - move `formData.isPublic ? 'public' : 'private'` to separate var

File: src/ducks/Signals/signalModal/SignalMasterModalForm.js [hunk 5]
```diff
@@ -49,32 +55,60 @@ const SignalMasterModalForm = ({
...
+                  {dialogTitle}
+                  {isShared && <div className={styles.shared}>Shared</div>}
+                </>
+              )
+            }
+            classes={styles}
+          >
+            <Dialog.ScrollContent className={styles.TriggerPanel}>
+              {isLoading && <div className={styles.loading}>Loading...</div>}
+              {!isLoading && (
```
Comment (haritonasty) [inline src/ducks/Signals/signalModal/SignalMasterModalForm.js]: Can we use animation component for Loading state? `Loader` from `sun-ui` for ex.

File: src/ducks/Signals/signalModal/SignalMasterModalForm.js [hunk 6]
```diff
@@ -49,32 +55,60 @@ const SignalMasterModalForm = ({
...
+        if (isError) {
+          throw new Error(`Can't find such public trigger with id ${triggerId}`)
+        }
+
+        return (
+          <Dialog
+            open={dialogOpenState}
+            onOpen={() => {
+              setDialogOpenState(true)
+            }}
```
Comment (haritonasty) [inline src/ducks/Signals/signalModal/SignalMasterModalForm.js]: fold `LOC 74 - 76` to 1 line removing brackets in callback

File: src/ducks/Signals/signalModal/SignalMasterModalForm.module.scss [hunk 7]
```diff
@@ -19,3 +19,31 @@
...
+}
+
+.shared {
+  @include text('body-3');
+
+  display: inline-block;
+  padding: 5px 10px;
+  color: var(--white);
+  background-color: var(--jungle-green);
+  border-radius: 4px;
```
Comment (haritonasty) [inline src/ducks/Signals/signalModal/SignalMasterModalForm.module.scss]: looks like `Button accent='positive' variant='fill'`. Is it possible to change?

File: src/ducks/Signals/common/getSignal.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
+import { graphql } from 'react-apollo'
+import { compose } from 'recompose'
+import { TRIGGER_BY_ID_QUERY } from './queries'
+import { mapGQLTriggerToProps } from '../utils/utils'
+import * as qs from 'query-string'
+
```
Comment (haritonasty) [inline src/ducks/Signals/common/getSignal.js]: `qs` import should be before internal imports

File: src/ducks/Signals/signalFormManager/aboutForm/AboutForm.js [hunk 9]
```diff
@@ -38,74 +47,76 @@ const AboutForm = ({ triggerMeta, onSubmit, onBack }) => {
...
-                  onChange={value =>
-                    setTrigger({
-                      ...trigger,
-                      title: value
-                    })
-                  }
-                />
+      {({ values: { description = '' }, isSubmitting, isValid }) => {
+        return (
+          <Form className={styles.AboutForm}>
```
[the remark this answers is not in the document]
Comment (GarageInc) [reply src/ducks/Signals/signalFormManager/aboutForm/AboutForm.js]: But "return" is better for debugging :)

File: src/ducks/Signals/signalModal/SignalMasterModalForm.js (hunk 5 above)
Comment (GarageInc) [reply src/ducks/Signals/signalModal/SignalMasterModalForm.js]: Replaced

File: src/ducks/Signals/signalFormManager/aboutForm/AboutForm.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (haritonasty) [reply src/ducks/Signals/signalFormManager/aboutForm/AboutForm.js]: oh, ok)

**Pull request merged**